examples/02_first_level_models/plot_fixed_effects.py

```diff
@@ -0,0 +1,112 @@
+"""
+Example of explicit fixed effects fMRI model fitting
+====================================================
+
+This example illustrates how to 
+
+For details on the data, please see:
+
+Dehaene-Lambertz G, Dehaene S, Anton JL, Campagne A, Ciuciu P, Dehaene
+G, Denghien I, Jobert A, LeBihan D, Sigman M, Pallier C, Poline
+JB. Functional segregation of cortical language areas by sentence
+repetition. Hum Brain Mapp. 2006: 27:360--371.
+http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2653076#R11
+
+Please see `plot_fiac_analysis.py` example for details.  The main
+difference is that the fixed-effects model is run explicitly here,
+after GLM fitting on two sessions.
+
+"""
+
+###############################################################################
+# Create a write directory to work
+# it will be a 'results' subdirectory of the current directory.
+from os import mkdir, path, getcwd
+write_dir = path.join(getcwd(), 'results')
+if not path.exists(write_dir):
+    mkdir(write_dir)
+
+#########################################################################
+# Prepare data and analysis parameters
+# --------------------------------------
+# 
+# Note that there are two sessions
+
+from nistats import datasets
+data = datasets.fetch_fiac_first_level()
+fmri_img = [data['func1'], data['func2']]
+
+#########################################################################
+# Create a mean image for plotting purpose
+from nilearn.image import mean_img
+mean_img_ = mean_img(fmri_img[0])
+
+#########################################################################
+# The design matrices were pre-computed, we simply put them in a list of DataFrames
+design_files = [data['design_matrix1'], data['design_matrix2']]
+import pandas as pd
+import numpy as np
+design_matrices = [pd.DataFrame(np.load(df)['X']) for df in design_files]
+
+#########################################################################
+# GLM estimation
+# ----------------------------------
+# GLM specification. Note that the mask was provided in the dataset. So we use it.
+
+from nistats.first_level_model import FirstLevelModel
+fmri_glm = FirstLevelModel(mask_img=data['mask'], smoothing_fwhm=5,
+                           minimize_memory=True)
+
+#########################################################################
+# Compute fixed effects of the two runs and compute related images
+# For this, we first define the contrasts as we would do for a single session
+n_columns = design_matrices[0].shape[1]
+
+def pad_vector(contrast_, n_columns):
```

Comment:
this is used only once in the example, so maybe it can be inlined and avoid the function definition

Reply:
I like using the function, I am wondering if we should include this as an automatic feature at some point in the future because many examples seem to need padding.